### PR TITLE
CLI-604: Drop support for PHP 7.3

### PR DESCRIPTION
**Pull request by anavarre**
**Motivation**

PHP 7.3 was recently EOL'ed https://www.php.net/supported-versions.php

**Proposed changes**

Support PHP 7.4+ only.

**Alternatives considered**
N/A

**Testing steps**
Confirm ACLI works well both with PHP 7.4 and 8.0, especially with CI.

1. Follow the [contribution guide](https://github.com/acquia/cli/blob/master/CONTRIBUTING.md#building-and-testing) to set up your development environment.
2. Clear the kernel cache to pick up new and changed commands: `./bin/acli ckc`
3. (add specific steps for this pr)

**Merge requirements**
- [ ] _Bug_, _enhancement_, or _breaking change_ label applied
- [ ] Manual testing by a reviewer

### CODE REVIEW

File: composer.lock [hunk 1]
```diff
@@ -4,7 +4,7 @@
         "Read more about it at https://getcomposer.org/doc/01-basic-usage.md#installing-dependencies",
         "This file is @generated automatically"
     ],
-    "content-hash": "eb51104c773c3d968b72b17826961fab",
+    "content-hash": "496f231ca5f6ec3243a5b2ccb6b1a2cb",
```
Comment (anavarre) [inline composer.lock:7]: I _think_ a `composer update` was required here so `composer.lock` would be updated.

**Comment by anavarre** [comment]
CI failing with

```
PHP CodeSniffer Config installed_paths set to ../../acquia/coding-standards/src/Standards,../../drupal/coder/coder_sniffer,../../phpcompatibility/php-compatibility,../../sirbrillig/phpcs-variable-analysis,../../slevomat/coding-standard
Watch out! GrumPHP is sniffing your commits!
phpstan/extension-installer: Extensions installed
> phpstan/phpstan-deprecation-rules: installed
dealerdirect/phpcodesniffer-composer-installer contains a Composer plugin which is currently not in your allow-plugins config. See https://getcomposer.org/allow-plugins
Do you trust "dealerdirect/phpcodesniffer-composer-installer" to execute code and wish to enable it now? (writes "allow-plugins" to composer.json) [y,n,d,?] 
                                                          
  [Symfony\Component\Console\Exception\RuntimeException]  
  Aborted                                                 
                                                          

Error: Process completed with exit code 1.
```

**Comment by grasmash** [comment]
That’s a new composer 2.2 thing. You probably need to run this locally with 2.2, answer the question, and it will write the new required configuration to the composer.json file.

**Comment by anavarre** [comment]
That was it, thank you! Also 🎅🎄🎁

**Pull request merged**